endpoints/openrtb2/auction.go

```diff
@@ -964,3 +997,13 @@ func writeError(errs []error, w http.ResponseWriter) bool {
 	}
 	return false
 }
+
+// Checks to see if an error in an error list is a fatal error
+func fatalError(errL []error) bool {
+	for _, err := range errL {
+		if _, ok := err.(*errortypes.NonFatalValidation); !ok {
```

Comment:
I stumbled on this interesting blog post on error handling a little while back: https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully

See the sections titled "Problems with error types", "Conclusion: avoid error types", and "Assert errors for behaviour, not type"

Reply:
Hah, we happen to be 90% there already ... well I'll do it by importing `errortypes`, but if it ever becomes an issue we can copy a few lines from there and break the dependence `errortypes`